src/qibolab/instruments/qm/config/pulses.py

```diff
@@ -24,6 +25,25 @@ def operation(pulse):
     return str(hash(pulse))
 
 
+def baked_duration(duration: int) -> int:
+    if duration < 16:
+        return 16
+    elif duration % 4 != 0:
+        return duration - (duration % 4) + 4
+    return duration
+
+
+def bake(waveforms: npt.NDArray) -> npt.NDArray:
+    """Handle waveforms with length that is not multiple of 4."""
+    ncomponents, duration = waveforms.shape
+    new_duration = baked_duration(duration)
+    if new_duration == duration:
+        return waveforms
+    pad_len = new_duration - duration
+    padding = np.zeros((ncomponents, pad_len), dtype=waveforms.dtype)
+    return np.concatenate((waveforms, padding), axis=1)
```

Comment:
[`np.pad`](https://numpy.org/doc/stable/reference/generated/numpy.pad.html)

Reply:
Thanks, I didn't think to look for this. Fixed in 7e03145c7b60b98c3a7e13300b6411b6722a074f.